libopenage/game_control.cpp

```diff
@@ -562,6 +568,9 @@ void GameControl::set_mode(int mode_index) {
 			engine->get_input_manager().remove_context(this->active_mode);
 
 			// set the new active mode
+			if (this->active_mode) {
```

Comment:
> can you change this to a check to test if the previous mode is not the new mode?
> that way, a non-change in modes doesn't call the exit.

It will not call `on_exit()`, but it will call `on_enter()`.
Maybe I should check if the previous mode is not the new mode in this if statement `if (mode_index < std::distance(std::begin(this->modes), std::end(this->modes)) &&this->modes[mode_index]->available())`. that way, non of these functions will be called if the mode is not being changed.
What do you think ?